### PR TITLE
unit test for go/yaml2/yaml.go

### PR DESCRIPTION
**Pull request by Its-Maniaco**
Partial Fix to #14931 
Covers unit test for `go/yaml2/yaml.go`

image

The test file is not testing any function.

### CODE REVIEW

File: go/yaml2/yaml_test.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
+package yaml2
+
```
Comment (frouioui) [inline go/yaml2/yaml_test.go:2]: License header is missing

**Comment by frouioui** [comment]
> The code coverage is coming out to be 0%, I think it is because the test is not directly testing any function.

Right, it is only covering variable declarations which I do not think count in the coverage report. I think it is fine as it is.

File: go/yaml2/yaml_test.go (hunk 1 above)
Comment (Its-Maniaco) [reply go/yaml2/yaml_test.go:15]: Added and with correct year this time.

File: go/yaml2/yaml_test.go [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+	//testing Marshal
+	marshaledData, err := Marshal(inputData)
+	assert.NoError(t, err)
+
+	//testing Unmarshal
+	var unmarshaledData TestStruct
+	err = Unmarshal(marshaledData, &unmarshaledData)
+	assert.NoError(t, err)
+	if !reflect.DeepEqual(inputData, unmarshaledData) {
```
Comment (dbussink) [inline go/yaml2/yaml_test.go:46]: `assert.Equal` afaik ends up doing a deep equal too, so let's use that.

Comment (Its-Maniaco) [reply go/yaml2/yaml_test.go:46]: Yeah [this](https://github.com/stretchr/testify/blob/v1.8.4/assert/assertions.go#L58) does use deep equal. Will change it.

File: go/yaml2/yaml_test.go [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+	}
+
+	inputData := TestStruct{
+		"tricky text to test text",
+		32,
+		true,
+		3.141,
+	}
+
+	//testing Marshal
```
Comment (dbussink) [inline go/yaml2/yaml_test.go:38]: Should also have a test per function here.

Comment (Its-Maniaco) [reply go/yaml2/yaml_test.go:38]: Can you please elaborate? Would it be better to Marshal and Unmarshal for each struct field or assert and verify after marshaling and then doing same for unmarshaling?

File: go/yaml2/yaml_test.go [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package yaml2
+
+import (
+	"reflect"
```
Comment (ajm188) [inline go/yaml2/yaml_test.go:17]: this needs to be removed

**Review by GuptaManan100 — APPROVED** [review]

**Review by harshit-gangal — APPROVED** [review]

**Pull request merged**